### PR TITLE
KC aggregate.

### PR DESCRIPTION
**Pull request by lukaDoric**

### CODE REVIEW

File: src/Tutor.Core/DomainModel/KnowledgeComponents/AssessmentEventSelector.cs [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
 using System.Linq;
+using FluentResults;
 using Tutor.Core.DomainModel.AssessmentEvents;
-using Tutor.Core.DomainModel.KnowledgeComponents;
 
-namespace Tutor.Core.InstructorModel.Instructors
+namespace Tutor.Core.DomainModel.KnowledgeComponents
 {
-    public class DefaultInstructor : IInstructor
+    public class AssessmentEventSelector : IAssessmentEventSelector
```
Comment (Luburic) [inline src/Tutor.Core/DomainModel/KnowledgeComponents/AssessmentEventSelector.cs:8]: AESelector is generic and this class is not. LeastCorrectAssessmentEventSelector?

File: src/Tutor.Core/DomainModel/KnowledgeComponents/KnowledgeComponentMastery.cs [hunk 2]
```diff
@@ -1,21 +1,48 @@
...
+            KnowledgeComponent = knowledgeComponent;
+        }
+
+        public void UpdateKcMastery(Submission submission, IKCRepository kcRepository)
+        {
+            var submissions = KnowledgeComponent.AssessmentEvents
+                .FirstOrDefault(ae => ae.Id == submission.AssessmentEventId)?.Submissions;
+            var currentCorrectnessLevel = 0.0;
+
+            if (submissions?.Any() == true)
```
[the remark this answers is not in the document]
Comment (lukaDoric) [reply src/Tutor.Core/DomainModel/KnowledgeComponents/KnowledgeComponentMastery.cs:30]: image

File: src/Tutor.Core/DomainModel/KnowledgeComponents/KnowledgeComponent.cs [hunk 3]
```diff
@@ -10,6 +11,7 @@ public class KnowledgeComponent
         public string Description { get; private set; }
 
         public List<KnowledgeComponent> KnowledgeComponents { get; private set; }
-        public List<KnowledgeComponentMastery> Masteries { get; private set; }
+        public List<KnowledgeComponentMastery> KnowledgeComponentMasteries { get; private set; }
```
[the remark this answers is not in the document]
Comment (milana-todorovic) [reply src/Tutor.Core/DomainModel/KnowledgeComponents/KnowledgeComponent.cs:14]: I agree that this property doesn't seem necessary at this point. I'm not entirely sure how the current version of EFCore handles one to many associations though, is just keeping the property that references the KC from KCMastery enough for EFCore to correctly interpret the association?

File: src/Tutor.Core/DomainModel/AssessmentEvents/AssessmentEvent.cs [hunk 4]
```diff
@@ -18,5 +19,10 @@ protected AssessmentEvent(int id, int knowledgeComponentId)
         }
 
         public abstract Evaluation EvaluateSubmission(Submission submission);
+
+        public double GetMaximumSubmissionCorrectness()
+        {
+            return Submissions.Any() ? Submissions.OrderBy(sub => sub.CorrectnessLevel).Last().CorrectnessLevel : 0.0;
```
[the remark this answers is not in the document]
Comment (lukaDoric) [reply src/Tutor.Core/DomainModel/AssessmentEvents/AssessmentEvent.cs:25]: It contains submission for specific learner. We fetch everything here (SubmissionService):
image
Repo:
image

**Review by Luburic — APPROVED** [review]
Looks good. Feel free to merge and delete branch if you are done.

**Pull request merged**